sources/api/migration/migrator/src/main.rs

```diff
@@ -328,8 +426,112 @@ where
 ///
 /// The given data store is used as a starting point; each migration is given the output of the
 /// previous migration, and the final output becomes the new data store.
-fn run_migrations<P1, P2>(
+fn run_migrations<P, S>(
+    repository: &tough::Repository<'_, tough::FilesystemTransport>,
     direction: Direction,
+    migrations: &[S],
+    source_datastore: P,
+    new_version: &Version,
+) -> Result<PathBuf>
+where
+    P: AsRef<Path>,
+    S: AsRef<str>,
+{
+    // We start with the given source_datastore, updating this after each migration to point to the
+    // output of the previous one.
+    let mut source_datastore = source_datastore.as_ref();
+    // We create a new data store (below) to serve as the target of each migration.  (Start at
+    // source just to have the right type; we know we have migrations at this point.)
+    let mut target_datastore = source_datastore.to_owned();
+    // Any data stores we create that aren't the final one, i.e. intermediate data stores, will be
+    // removed at the end.  (If we fail and return early, they're left for debugging purposes.)
+    let mut intermediate_datastores = HashSet::new();
+
+    for migration in migrations {
+        let migration = migration.as_ref();
+        // get the migration from the repo
+        let lz4_bytes = repository
+            .read_target(migration)
+            .context(error::LoadMigration { migration })?
+            .context(error::MigrationNotFound { migration })?;
+
+        // deflate with an lz4 decoder read
+        let mut reader = lz4::Decoder::new(lz4_bytes).context(error::Lz4Decode { migration })?;
+
+        // create a sealed command with pentacle
```

Comment:
```suggestion
        // Create a sealed command with pentacle, so we can run the verified bytes from memory
```

Reply:
Addressed in [4f34544](https://github.com/bottlerocket-os/bottlerocket/compare/afc4b5d9ddfb7c6fb24f30abdaf6a5a000df222c..4f345446016e314c3da3f0855c92db5ca979f2a4)